File: frontend/src/types/migration.ts
export interface TargetCloudFlowAction {
  id: string;
  name: string;
  type: string;
  connector?: string;
  operationId?: string;
  parameters: Record<string, any>;
  runAfter: Record<string, string[]>;
}

export interface TargetCloudFlow {
  name: string;
  description: string;
  trigger: Record<string, any>;
  actions: TargetCloudFlowAction[];
  connectionsNeeded: string[];
  environmentVariables: string[];
}

export interface TargetDesktopFlowAction {
  id: string;
  name: string;
  module: string;
  statement: string;
  parameters: Record<string, any>;
}

export interface TargetDesktopFlow {
  name: string;
  description: string;
  subroutines: string[];
  inputVariables: string[];
  outputVariables: string[];
  actions: TargetDesktopFlowAction[];
  prerequisites: string[];
}

export interface MigrationRoadmapPhase {
  phase: string;
  durationDays: number;
  tasks: string[];
}

export interface TargetArchitecture {
  architectureType: string;
  summary: string;
  orchestrationPattern: string;
  cloudFlows: TargetCloudFlow[];
  desktopFlows: TargetDesktopFlow[];
  recommendedConnections: string[];
  securityAndCredentialsGuidance: string;
  migrationRoadmapPhases: MigrationRoadmapPhase[];
}

export interface MigrationRisk {
  severity: 'Low' | 'Medium' | 'High';
  title: string;
  description: string;
  mitigation: string;
}

export interface ManualReviewItem {
  step: number;
  command: string;
  task: string;
  file: string;
  raw: Record<string, any>;
  reason: string;
  suggestedAction: string;
}

export interface MigrationPlanModel {
  job_id: string;
  workflow_name: string;
  architecture: TargetArchitecture;
  action_plans: Record<string, any>[];
  migration_risks: MigrationRisk[];
  manual_review_items: ManualReviewItem[];
  estimated_effort_hours: number;
}

File: backend/models/migration.py
from typing import List, Dict, Any, Optional
from pydantic import BaseModel, Field

class TargetCloudFlowAction(BaseModel):
    id: str
    name: str
    type: str  # e.g., "OpenApiConnection", "Scope", "If", "Foreach", "DesktopFlow"
    connector: Optional[str] = None
    operationId: Optional[str] = None
    parameters: Dict[str, Any] = Field(default_factory=dict)
    runAfter: Dict[str, List[str]] = Field(default_factory=dict)

class TargetCloudFlow(BaseModel):
    name: str
    description: str = ""
    trigger: Dict[str, Any] = Field(default_factory=lambda: {"type": "Recurrence", "recurrence": {"frequency": "Day", "interval": 1}})
    actions: List[TargetCloudFlowAction] = Field(default_factory=list)
    connectionsNeeded: List[str] = Field(default_factory=list)
    environmentVariables: List[str] = Field(default_factory=list)

class TargetDesktopFlowAction(BaseModel):
    id: str
    name: str
    module: str  # e.g., "Excel", "WebAutomation", "UIAutomation", "File"
    statement: str
    parameters: Dict[str, Any] = Field(default_factory=dict)

class TargetDesktopFlow(BaseModel):
    name: str
    description: str = ""
    subroutines: List[str] = Field(default_factory=list)
    inputVariables: List[str] = Field(default_factory=list)
    outputVariables: List[str] = Field(default_factory=list)
    actions: List[TargetDesktopFlowAction] = Field(default_factory=list)
    prerequisites: List[str] = Field(default_factory=list)

class TargetArchitecture(BaseModel):
    architectureType: str  # "Cloud Only", "Desktop Only", "Hybrid (Cloud Orchestrated Desktop)"
    summary: str
    orchestrationPattern: str
    cloudFlows: List[TargetCloudFlow] = Field(default_factory=list)
    desktopFlows: List[TargetDesktopFlow] = Field(default_factory=list)
    recommendedConnections: List[str] = Field(default_factory=list)
    securityAndCredentialsGuidance: str
    migrationRoadmapPhases: List[Dict[str, Any]] = Field(default_factory=list)

class MigrationPlanModel(BaseModel):
    job_id: str
    workflow_name: str
    architecture: TargetArchitecture
    action_plans: List[Dict[str, Any]] = Field(default_factory=list)
    migration_risks: List[Dict[str, Any]] = Field(default_factory=list)
    manual_review_items: List[Dict[str, Any]] = Field(default_factory=list)
    estimated_effort_hours: int = 0
